### PR TITLE
[GEOT-5932]:GridCoverageRenderer set parent coverage property

### PR DESCRIPTION
**Pull request by dromagnoli**

### CODE REVIEW

File: modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/GridCoverageRenderer.java [hunk 1]
```diff
@@ -748,7 +743,18 @@ private GridCoverage2D renderCoverage(final GridCoverage2D gridCoverage,
...
     }
 
-
+	private RenderedImage getImageFromParentCoverage(GridCoverage2D parentCoverage) {
+		if (parentCoverage == null) {
+			return null;
+		}
+		RenderedImage ri = parentCoverage.getRenderedImage();
+		if (ri != null) {
+			PlanarImage pi = PlanarImage.wrapRenderedImage(ri);
```
Comment (aaime) [inline modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/GridCoverageRenderer.java]: This already checks if ri is a planar and avoids double wrapping, right?

File: modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/GridCoverageRenderer.java [hunk 2]
```diff
@@ -748,7 +743,18 @@ private GridCoverage2D renderCoverage(final GridCoverage2D gridCoverage,
...
+			pi.setProperty("ParentCoverage", parentCoverage);
```
Comment (aaime) [inline modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/GridCoverageRenderer.java]: Best to put this name in a public constant for clients and tests to reuse

File: modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/GridCoverageRenderer.java (hunk 1 above)
Comment (dromagnoli) [reply modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/GridCoverageRenderer.java]: Yep. wrapRenderedImage avoid double wrapping by simply returning the input object if already a PlanarImage.

File: modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/GridCoverageRenderer.java (hunk 2 above)
Comment (dromagnoli) [reply modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/GridCoverageRenderer.java]: Right. I forgot to do that. Fixing...

**Comment by dromagnoli** [comment]
@aaime Feedbacks addressed (putting the property as a constant for re-use)

**Pull request merged**